### PR TITLE
Loosen pyarrow requirements to include 1.x

### PR DESCRIPTION
**Pull request by fhoehle**
This PR replaces #328 and addresses the requested changes.

- [x] Changelog entry

### CODE REVIEW

File: tests/serialization/test_arrow_compat.py [hunk 1]
```diff
@@ -48,7 +48,10 @@ def test_arrow_compat(arrow_version, reference_store, mocker):
         store=reference_store, key=arrow_version + ".parquet", date_as_object=True
     )
 
-    if arrow_version in ("0.14.1", "0.15.0", "0.16.0") and not ARROW_LARGER_EQ_0141:
+    if (
+        arrow_version in ("0.14.1", "0.15.0", "0.16.0", "1.0.0")
+        and not ARROW_LARGER_EQ_0141
```
[the remark this answers is not in the document]
Comment (fhoehle) [reply tests/serialization/test_arrow_compat.py:53]: @fjetter kind bump.

[the remark this answers is not in the document]
Comment (fjetter) [reply tests/serialization/test_arrow_compat.py:53]: My point is `ARROW_LARGER_EQ_0141` is always true for `arrow_version in ("0.14.1", "0.15.0", "0.16.0", "1.0.0")`
therefore we'll have either

`True and not True` for arrow >= 0.14.1

or

`False and not False` for arrow==0.13.0

therefore this statement is always `False`. Am I missing something here?

[the remark this answers is not in the document]
Comment (fjetter) [reply tests/serialization/test_arrow_compat.py:53]: If it is always false, we can as well remove it

**Review by fjetter — APPROVED** [review]

File: tests/serialization/test_arrow_compat.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (fjetter) [reply tests/serialization/test_arrow_compat.py:53]: nvm. We'll drop these versions anyhow soon. Sorry for blocking

**Comment by fhoehle** [comment]
@fjetter thanks! Would it be possible to publish a release within next days?

**Pull request merged**